os/fs/semaphore/sem_close.c

```diff
@@ -159,6 +158,10 @@ int sem_close(FAR sem_t *sem)
 
 	inode_semgive();
 	return OK;
+
+	#endif							/* CONFIG_FS_NAMED_SEMAPHORES */
+	set_errno(ENOTSUP);
+	return ENOTSUP;
```

Comment:
The function description says 
```
 * Return Value:
 *  0 (OK), or -1 (ERROR) if unsuccessful.
```
There is no `ENOTSUP`
Let's return ERROR only with errno ENOTSUP.